### PR TITLE
[Candidate] Add polls, create national and local polls

### PR DESCRIPTION
**Pull request by illusionOfParadise**

### CODE REVIEW

File: templates/poll/local_list.html.twig [hunk 1]
```diff
@@ -0,0 +1,94 @@
+{% extends 'poll/_layout.html.twig' %}
+
+{% block javascripts %}
+    <script type="text/javascript">
+    </script>
+{% endblock %}
```
Comment (ottaviano) [inline templates/poll/local_list.html.twig:6]:
```suggestion
```

File: templates/poll/local_list.html.twig [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                            <td>{{ item.yes_count }}</td>
+                            <td>{{ item.no_count }}</td>
+                            <td>
+                                <span class="adherent-dashboard-tag">{{ ('geo_zone.'~poll.zone.type)|trans }}</span><br>
+                                <div class="b__nudge--top-10">{{ poll.zone }}</div>
+                            </td>
+                            <td>{{ poll.author ? poll.author.fullName : 'Ancien adhérent' }}</td>
+                            <td>
+                                <span class="status status__{{ poll.published ? '1' : '2' }}">
+                                    {{ poll.published ? 'Publié' : 'Non publié' }}
```
Comment (ottaviano) [inline templates/poll/local_list.html.twig:56]:
```suggestion
                                    {{ poll.published ? 'Publiée' : 'Non publiée' }}
```

File: templates/poll/local_list.html.twig [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                                        }
+                                    }
+                                ] %}
+
+                                {% include 'components/_annimated_sub_menu.html.twig' with {items: items} %}
+                            </td>
+                        </tr>
+                    {% else %}
+                        <tr>
+                            <td colspan="8" class="text--center">
```
Comment (ottaviano) [inline templates/poll/local_list.html.twig:83]:
```suggestion
                            <td colspan="10" class="text--center">
```

File: templates/poll/create.html.twig [hunk 4]
```diff
@@ -0,0 +1,60 @@
+{% extends 'poll/_layout.html.twig' %}
+
+{% block stylesheets %}
+    <link rel="stylesheet" href="{{ asset('bundles/sonataadmin/vendor/jqueryui/themes/base/minified/jquery-ui.min.css') }}" />
```
Comment (ottaviano) [inline templates/poll/create.html.twig:4]: do we need this CSS file here?

File: templates/poll/create.html.twig [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+{% endblock %}
+
+{% block polls_content %}
+    {% form_theme form with ['jecoute/_form_theme.html.twig'] %}
```
Comment (ottaviano) [inline templates/poll/create.html.twig:8]: it's used?

File: src/Controller/EnMarche/Poll/PollCandidateController.php [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+class PollCandidateController extends AbstractPollController
+{
+    protected function getSpaceName(): string
+    {
+        return JecouteSpaceEnum::CANDIDATE_SPACE;
+    }
+
+    protected function getZones(Adherent $adherent): array
+    {
+        return [$adherent->getCandidateManagedArea()->getZone()];
```
[the remark this answers is not in the document]
Comment (illusionOfParadise) [reply src/Controller/EnMarche/Poll/PollCandidateController.php:24]: It's already done in `AbstractPollController`.

File: src/Entity/Poll/NationalPoll.php [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+use ApiPlatform\Core\Annotation\ApiResource;
+use App\Entity\Administrator;
+use Doctrine\ORM\Mapping as ORM;
+use Ramsey\Uuid\UuidInterface;
+
+/**
+ * @ApiResource(
+ *     collectionOperations={
+ *         "get": {
+ *             "path": "/polls",
```
[the remark this answers is not in the document]
Comment (illusionOfParadise) [reply src/Entity/Poll/NationalPoll.php:14]: URL is already used, we don't change it at the moment.

File: src/Repository/Poll/PollRepository.php [hunk 8]
```diff
@@ -4,12 +4,21 @@
 
 use App\Entity\Poll\Poll;
 use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
-use Doctrine\Persistence\ManagerRegistry;
 
-class PollRepository extends ServiceEntityRepository
+abstract class PollRepository extends ServiceEntityRepository
```
Comment (ottaviano) [inline src/Repository/Poll/PollRepository.php:8]: `AbstractPollRepository`?

**Review by ottaviano — APPROVED** [review]

**Pull request merged**